core/src/scrim/scrim.resolver.ts

```diff
@@ -27,7 +28,17 @@ export class ScrimResolver {
         private readonly gameModeService: GameModeService,
     ) {}
 
-    @ResolveField()
+    @ResolveField(() => GameMode)
+    async gameMode(@Root() scrim: Scrim): Promise<GameMode> {
+        return this.gameModeService.getGameModeById(scrim.gameModeId);
```

Comment:
We should be checking to see if it already exists on the root object to prevent double fetching